pkgs/java_http/lib/src/java_client.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:http/http.dart';
+import 'package:jni/jni.dart';
+import 'package:path/path.dart';
+
+import 'third_party/java/net/URL.dart';
+
+class JavaClient extends BaseClient {
+  void _initJVM() {
+    if (!Platform.isAndroid) {
+      Jni.spawnIfNotExists(dylibDir: join('build', 'jni_libs'));
+    }
+  }
+
+  @override
+  Future<StreamedResponse> send(BaseRequest request) async {
+    _initJVM();
```

Comment:
Would it make sense to do this in the constructor so it is only done once per class (and you don't get an unexpected exception later on in your application run)? Feel free to fix in a follow-up PR.

Reply:
Yeah you're right, it definitely makes more sense to put the call to `_initJVM()` in the constructor.
I'll make the change in a follow-up PR and I've added a TODO comment for now to remind me.